paginator/is_nil_test.go

```diff
@@ -0,0 +1,53 @@
+package paginator
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_IsNil(t *testing.T) {
+	t.Parallel()
+
+	var testErrNilPtr *testError
+	var testMapNilPtr map[string]struct{}
+	var testArrayNilPtr *[1]string
+	var testChanNilPtr *chan string
+	var testSliceNilPtr *[]string
+
+	var testChanString chan string
+
+	tc := []struct {
+		i    interface{}
+		want bool
+	}{
+		{i: &testError{}, want: false},
+		{i: testError{}, want: false},
+		{i: &map[string]struct{}{}, want: false},
+		{i: map[string]struct{}{}, want: false},
+		{i: [1]string{}, want: false},
+		{i: &[1]string{}, want: false},
```

Comment:
Just wondering that are these two cases causing the old version to break?

Reply:
Just the first test: `{i: [1]string{}, want: false}` causes the panic.  The next test is a ptr to array and that's okay with the original version.